src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Wrong side parameter(s)');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Wrong side parameter(s)');
+    }
+  }
 
+  getArea(): number {
+    const halfPerimeter: number = (this.a + this.b + this.c) / 2;
+
+    const area: number = Math.sqrt(halfPerimeter * (halfPerimeter - this.a)
+      * (halfPerimeter - this.b) * (halfPerimeter - this.c));
+
+    return Math.floor((area) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+  ) {
+    if (a <= 0) {
+      throw new Error('Wrong radius');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor((Math.PI * (this.a ** 2)) * 100) / 100;
```

Comment:
You can create a helper for this calculation as you do the same twice. `Math.floor((area) * 100) / 100`